src/node.cc

```diff
@@ -1510,15 +1511,16 @@ void AppendExceptionLine(Environment* env,
 
   Local<String> arrow_str = String::NewFromUtf8(env->isolate(), arrow);
 
-  if (!arrow_str.IsEmpty() && !err_obj.IsEmpty() && err_obj->IsNativeError()) {
+  if (!arrow_str.IsEmpty() && !err_obj.IsEmpty() &&
+      (!handlingFatalError || err_obj->IsNativeError())) {
```

Comment:
If I got the semantics wrong, then that's probably good indication that the current clause is hard to understand. :-)